### PR TITLE
1167 fixes protobuf

### LINKED ISSUE #1167: from torch.utils.tensorboard import SummaryWriter  Descriptors cannot not be created directly.
system protobuf version was 3.20.1, but it's updated when running some notebooks

```
03:16:20  Running ./3d_segmentation/unet_segmentation_3d_ignite.ipynb
03:16:20  Checking PEP8 compliance...
03:16:20  Running notebook...
03:16:20  Before:
03:16:20      "max_epochs = 10\n",
03:16:20  After:
03:16:20      "max_epochs = 1\n",
03:16:26  MONAI version: 1.1.0+22.gf14d50ab
03:16:26  Numpy version: 1.22.2
03:16:26  Pytorch version: 1.14.0a0+410ce96
03:16:26  MONAI flags: HAS_EXT = False, USE_COMPILED = False, USE_META_DICT = False
03:16:26  MONAI rev id: f14d50ab00baec9813c9061896cf7ab21c356891
03:16:26  MONAI __file__: /home/jenkins/agent/workspace/Monai-notebooks/MONAI/monai/__init__.py
03:16:26  
03:16:26  Optional dependencies:
03:16:26  Pytorch Ignite version: 0.4.10
03:16:26  Nibabel version: 5.0.0
03:16:26  scikit-image version: 0.19.3
03:16:26  Pillow version: 9.2.0
03:16:26  Tensorboard version: 2.9.0
03:16:26  gdown version: 4.6.0
03:16:26  TorchVision version: 0.15.0a0
03:16:26  tqdm version: 4.64.1
03:16:26  lmdb version: 1.4.0
03:16:26  psutil version: 5.9.4
03:16:26  pandas version: 1.5.2
03:16:26  einops version: 0.6.0
03:16:26  transformers version: 4.21.3
03:16:26  mlflow version: 2.1.1
03:16:26  pynrrd version: 1.0.0
03:16:26  
03:16:26  For details about installing the optional dependencies, please visit:
03:16:26      https://docs.monai.io/en/latest/installation.html#installing-the-recommended-dependencies
03:16:26  
03:16:26  papermill  --progress-bar -k python3
03:16:26  /usr/local/lib/python3.8/dist-packages/papermill/iorw.py:153: UserWarning: the file is not specified with any extension : -
03:16:26    warnings.warn(
03:16:50  
Executing:   0%|          | 0/29 [00:00<?, ?cell/s]
Executing:   3%|▎         | 1/29 [00:01<00:41,  1.47s/cell]
Executing:  10%|█         | 3/29 [00:07<01:08,  2.65s/cell]
Executing:  17%|█▋        | 5/29 [00:12<01:01,  2.57s/cell]
Executing:  38%|███▊      | 11/29 [00:20<00:30,  1.67s/cell]
Executing:  45%|████▍     | 13/29 [00:21<00:23,  1.46s/cell]
Executing:  66%|██████▌   | 19/29 [00:22<00:07,  1.29cell/s]
Executing:  66%|██████▌   | 19/29 [00:23<00:12,  1.26s/cell]
03:16:50  /usr/local/lib/python3.8/dist-packages/papermill/iorw.py:153: UserWarning: the file is not specified with any extension : -
03:16:50    warnings.warn(
03:16:50  Traceback (most recent call last):
03:16:50    File "/usr/local/bin/papermill", line 8, in <module>
03:16:50      sys.exit(papermill())
03:16:50    File "/usr/local/lib/python3.8/dist-packages/click/core.py", line 1130, in __call__
03:16:50      return self.main(*args, **kwargs)
03:16:50    File "/usr/local/lib/python3.8/dist-packages/click/core.py", line 1055, in main
03:16:50      rv = self.invoke(ctx)
03:16:50    File "/usr/local/lib/python3.8/dist-packages/click/core.py", line 1404, in invoke
03:16:50      return ctx.invoke(self.callback, **ctx.params)
03:16:50    File "/usr/local/lib/python3.8/dist-packages/click/core.py", line 760, in invoke
03:16:50      return __callback(*args, **kwargs)
03:16:50    File "/usr/local/lib/python3.8/dist-packages/click/decorators.py", line 26, in new_func
03:16:50      return f(get_current_context(), *args, **kwargs)
03:16:50    File "/usr/local/lib/python3.8/dist-packages/papermill/cli.py", line 250, in papermill
03:16:50      execute_notebook(
03:16:50    File "/usr/local/lib/python3.8/dist-packages/papermill/execute.py", line 128, in execute_notebook
03:16:50      raise_for_execution_errors(nb, output_path)
03:16:50    File "/usr/local/lib/python3.8/dist-packages/papermill/execute.py", line 232, in raise_for_execution_errors
03:16:50      raise error
03:16:50  papermill.exceptions.PapermillExecutionError: 
03:16:50  ---------------------------------------------------------------------------
03:16:50  Exception encountered at "In [9]":
03:16:50  ---------------------------------------------------------------------------
03:16:50  OptionalImportError                       Traceback (most recent call last)
03:16:50  Cell In[9], line 21
03:16:50       18 train_stats_handler.attach(trainer)
03:16:50       20 # TensorBoardStatsHandler plots loss at every iteration
03:16:50  ---> 21 train_tensorboard_stats_handler = TensorBoardStatsHandler(log_dir=log_dir, output_transform=lambda x: x)
03:16:50       22 train_tensorboard_stats_handler.attach(trainer)
03:16:50       24 # MLFlowHandler plots loss at every iteration on MLFlow web UI
03:16:50  
03:16:50  File /home/jenkins/agent/workspace/Monai-notebooks/MONAI/monai/handlers/tensorboard_handlers.py:131, in TensorBoardStatsHandler.__init__(self, summary_writer, log_dir, iteration_log, epoch_log, epoch_event_writer, epoch_interval, iteration_event_writer, iteration_interval, output_transform, global_epoch_transform, state_attributes, tag_name)
03:16:50       87 def __init__(
03:16:50       88     self,
03:16:50       89     summary_writer=None,
03:16:50     (...)
03:16:50      100     tag_name: str = DEFAULT_TAG,
03:16:50      101 ) -> None:
03:16:50      102     """
03:16:50      103     Args:
03:16:50      104         summary_writer: user can specify TensorBoard or TensorBoardX SummaryWriter,
03:16:50     (...)
03:16:50      129         tag_name: when iteration output is a scalar, tag_name is used to plot, defaults to ``'Loss'``.
03:16:50      130     """
03:16:50  --> 131     super().__init__(summary_writer=summary_writer, log_dir=log_dir)
03:16:50      132     self.iteration_log = iteration_log
03:16:50      133     self.epoch_log = epoch_log
03:16:50  
03:16:50  File /home/jenkins/agent/workspace/Monai-notebooks/MONAI/monai/handlers/tensorboard_handlers.py:51, in TensorBoardHandler.__init__(self, summary_writer, log_dir)
03:16:50       49 def __init__(self, summary_writer=None, log_dir: str = "./runs"):
03:16:50       50     if summary_writer is None:
03:16:50  ---> 51         self._writer = SummaryWriter(log_dir=log_dir)
03:16:50       52         self.internal_writer = True
03:16:50       53     else:
03:16:50  
03:16:50  File /home/jenkins/agent/workspace/Monai-notebooks/MONAI/monai/utils/module.py:418, in optional_import.<locals>._LazyRaise.__call__(self, *_args, **_kwargs)
03:16:50      413 def __call__(self, *_args, **_kwargs):
03:16:50      414     """
03:16:50      415     Raises:
03:16:50      416         OptionalImportError: When you call this method.
03:16:50      417     """
03:16:50  --> 418     raise self._exception
03:16:50  
03:16:50  File /home/jenkins/agent/workspace/Monai-notebooks/MONAI/monai/utils/module.py:370, in optional_import(module, version, version_checker, name, descriptor, version_args, allow_namespace_pkg, as_type)
03:16:50      368     actual_cmd = f"import {module}"
03:16:50      369 try:
03:16:50  --> 370     pkg = __import__(module)  # top level module
03:16:50      371     the_module = import_module(module)
03:16:50      372     if not allow_namespace_pkg:
03:16:50  
03:16:50  File /usr/local/lib/python3.8/dist-packages/torch/utils/tensorboard/__init__.py:12
03:16:50        9 del LooseVersion
03:16:50       10 del tensorboard
03:16:50  ---> 12 from .writer import FileWriter, SummaryWriter  # noqa: F401
03:16:50       13 from tensorboard.summary.writer.record_writer import RecordWriter  # noqa: F401
03:16:50  
03:16:50  File /usr/local/lib/python3.8/dist-packages/torch/utils/tensorboard/writer.py:9
03:16:50        6 import torch
03:16:50        8 from tensorboard.compat import tf
03:16:50  ----> 9 from tensorboard.compat.proto.event_pb2 import SessionLog
03:16:50       10 from tensorboard.compat.proto.event_pb2 import Event
03:16:50       11 from tensorboard.compat.proto import event_pb2
03:16:50  
03:16:50  File /usr/local/lib/python3.8/dist-packages/tensorboard/compat/proto/event_pb2.py:17
03:16:50       12 # @@protoc_insertion_point(imports)
03:16:50       14 _sym_db = _symbol_database.Default()
03:16:50  ---> 17 from tensorboard.compat.proto import summary_pb2 as tensorboard_dot_compat_dot_proto_dot_summary__pb2
03:16:50       20 DESCRIPTOR = _descriptor.FileDescriptor(
03:16:50       21   name='tensorboard/compat/proto/event.proto',
03:16:50       22   package='tensorboard',
03:16:50     (...)
03:16:50       26   ,
03:16:50       27   dependencies=[tensorboard_dot_compat_dot_proto_dot_summary__pb2.DESCRIPTOR,])
03:16:50       29 _WORKERHEALTH = _descriptor.EnumDescriptor(
03:16:50       30   name='WorkerHealth',
03:16:50       31   full_name='tensorboard.WorkerHealth',
03:16:50     (...)
03:16:50       55   serialized_end=1319,
03:16:50       56 )
03:16:50  
03:16:50  File /usr/local/lib/python3.8/dist-packages/tensorboard/compat/proto/summary_pb2.py:17
03:16:50       12 # @@protoc_insertion_point(imports)
03:16:50       14 _sym_db = _symbol_database.Default()
03:16:50  ---> 17 from tensorboard.compat.proto import tensor_pb2 as tensorboard_dot_compat_dot_proto_dot_tensor__pb2
03:16:50       20 DESCRIPTOR = _descriptor.FileDescriptor(
03:16:50       21   name='tensorboard/compat/proto/summary.proto',
03:16:50       22   package='tensorboard',
03:16:50     (...)
03:16:50       26   ,
03:16:50       27   dependencies=[tensorboard_dot_compat_dot_proto_dot_tensor__pb2.DESCRIPTOR,])
03:16:50       29 _DATACLASS = _descriptor.EnumDescriptor(
03:16:50       30   name='DataClass',
03:16:50       31   full_name='tensorboard.DataClass',
03:16:50     (...)
03:16:50       55   serialized_end=1228,
03:16:50       56 )
03:16:50  
03:16:50  File /usr/local/lib/python3.8/dist-packages/tensorboard/compat/proto/tensor_pb2.py:16
03:16:50       11 # @@protoc_insertion_point(imports)
03:16:50       13 _sym_db = _symbol_database.Default()
03:16:50  ---> 16 from tensorboard.compat.proto import resource_handle_pb2 as tensorboard_dot_compat_dot_proto_dot_resource__handle__pb2
03:16:50       17 from tensorboard.compat.proto import tensor_shape_pb2 as tensorboard_dot_compat_dot_proto_dot_tensor__shape__pb2
03:16:50       18 from tensorboard.compat.proto import types_pb2 as tensorboard_dot_compat_dot_proto_dot_types__pb2
03:16:50  
03:16:50  File /usr/local/lib/python3.8/dist-packages/tensorboard/compat/proto/resource_handle_pb2.py:16
03:16:50       11 # @@protoc_insertion_point(imports)
03:16:50       13 _sym_db = _symbol_database.Default()
03:16:50  ---> 16 from tensorboard.compat.proto import tensor_shape_pb2 as tensorboard_dot_compat_dot_proto_dot_tensor__shape__pb2
03:16:50       17 from tensorboard.compat.proto import types_pb2 as tensorboard_dot_compat_dot_proto_dot_types__pb2
03:16:50       20 DESCRIPTOR = _descriptor.FileDescriptor(
03:16:50       21   name='tensorboard/compat/proto/resource_handle.proto',
03:16:50       22   package='tensorboard',
03:16:50     (...)
03:16:50       26   ,
03:16:50       27   dependencies=[tensorboard_dot_compat_dot_proto_dot_tensor__shape__pb2.DESCRIPTOR,tensorboard_dot_compat_dot_proto_dot_types__pb2.DESCRIPTOR,])
03:16:50  
03:16:50  File /usr/local/lib/python3.8/dist-packages/tensorboard/compat/proto/tensor_shape_pb2.py:36
03:16:50       13 _sym_db = _symbol_database.Default()
03:16:50       18 DESCRIPTOR = _descriptor.FileDescriptor(
03:16:50       19   name='tensorboard/compat/proto/tensor_shape.proto',
03:16:50       20   package='tensorboard',
03:16:50     (...)
03:16:50       23   serialized_pb=_b('\n+tensorboard/compat/proto/tensor_shape.proto\x12\x0btensorboard\"{\n\x10TensorShapeProto\x12.\n\x03\x64im\x18\x02 \x03(\x0b\x32!.tensorboard.TensorShapeProto.Dim\x12\x14\n\x0cunknown_rank\x18\x03 \x01(\x08\x1a!\n\x03\x44im\x12\x0c\n\x04size\x18\x01 \x01(\x03\x12\x0c\n\x04name\x18\x02 \x01(\tB\x87\x01\n\x18org.tensorflow.frameworkB\x11TensorShapeProtosP\x01ZSgithub.com/tensorflow/tensorflow/tensorflow/go/core/framework/tensor_shape_go_proto\xf8\x01\x01\x62\x06proto3')
03:16:50       24 )
03:16:50       29 _TENSORSHAPEPROTO_DIM = _descriptor.Descriptor(
03:16:50       30   name='Dim',
03:16:50       31   full_name='tensorboard.TensorShapeProto.Dim',
03:16:50       32   filename=None,
03:16:50       33   file=DESCRIPTOR,
03:16:50       34   containing_type=None,
03:16:50       35   fields=[
03:16:50  ---> 36     _descriptor.FieldDescriptor(
03:16:50       37       name='size', full_name='tensorboard.TensorShapeProto.Dim.size', index=0,
03:16:50       38       number=1, type=3, cpp_type=2, label=1,
03:16:50       39       has_default_value=False, default_value=0,
03:16:50       40       message_type=None, enum_type=None, containing_type=None,
03:16:50       41       is_extension=False, extension_scope=None,
03:16:50       42       serialized_options=None, file=DESCRIPTOR),
03:16:50       43     _descriptor.FieldDescriptor(
03:16:50       44       name='name', full_name='tensorboard.TensorShapeProto.Dim.name', index=1,
03:16:50       45       number=2, type=9, cpp_type=9, label=1,
03:16:50       46       has_default_value=False, default_value=_b("").decode('utf-8'),
03:16:50       47       message_type=None, enum_type=None, containing_type=None,
03:16:50       48       is_extension=False, extension_scope=None,
03:16:50       49       serialized_options=None, file=DESCRIPTOR),
03:16:50       50   ],
03:16:50       51   extensions=[
03:16:50       52   ],
03:16:50       53   nested_types=[],
03:16:50       54   enum_types=[
03:16:50       55   ],
03:16:50       56   serialized_options=None,
03:16:50       57   is_extendable=False,
03:16:50       58   syntax='proto3',
03:16:50       59   extension_ranges=[],
03:16:50       60   oneofs=[
03:16:50       61   ],
03:16:50       62   serialized_start=150,
03:16:50       63   serialized_end=183,
03:16:50       64 )
03:16:50       66 _TENSORSHAPEPROTO = _descriptor.Descriptor(
03:16:50       67   name='TensorShapeProto',
03:16:50       68   full_name='tensorboard.TensorShapeProto',
03:16:50     (...)
03:16:50      100   serialized_end=183,
03:16:50      101 )
03:16:50      103 _TENSORSHAPEPROTO_DIM.containing_type = _TENSORSHAPEPROTO
03:16:50  
03:16:50  File /usr/local/lib/python3.8/dist-packages/google/protobuf/descriptor.py:560, in FieldDescriptor.__new__(cls, name, full_name, index, number, type, cpp_type, label, default_value, message_type, enum_type, containing_type, is_extension, extension_scope, options, serialized_options, has_default_value, containing_oneof, json_name, file, create_key)
03:16:50      554 def __new__(cls, name, full_name, index, number, type, cpp_type, label,
03:16:50      555             default_value, message_type, enum_type, containing_type,
03:16:50      556             is_extension, extension_scope, options=None,
03:16:50      557             serialized_options=None,
03:16:50      558             has_default_value=True, containing_oneof=None, json_name=None,
03:16:50      559             file=None, create_key=None):  # pylint: disable=redefined-builtin
03:16:50  --> 560   _message.Message._CheckCalledFromGeneratedFile()
03:16:50      561   if is_extension:
03:16:50      562     return _message.default_pool.FindExtensionByName(full_name)
03:16:50  
03:16:50  OptionalImportError: from torch.utils.tensorboard import SummaryWriter (Descriptors cannot not be created directly.
03:16:50  If this call came from a _pb2.py file, your generated code is out of date and must be regenerated with protoc >= 3.19.0.
03:16:50  If you cannot immediately regenerate your protos, some other possible workarounds are:
03:16:50   1. Downgrade the protobuf package to 3.20.x or lower.
03:16:50   2. Set PROTOCOL_BUFFERS_PYTHON_IMPLEMENTATION=python (but this will use pure-Python parsing and will be much slower).
03:16:50  
03:16:50  More information: https://developers.google.com/protocol-buffers/docs/news/2022-05-06#python-updates).
03:16:50  
03:16:50  For details about installing the optional dependencies, please visit:
03:16:50      https://docs.monai.io/en/latest/installation.html#installing-the-recommended-dependencies
```

### PR DESCRIPTION
**Pull request by wyli**
Fixes #1167

### Description
- pip install ray might upgrade protobuf automatically, but protobuf v4 is not compatible with some versions of pytorch's tensorboard summarywriter

### Checks

- [ ] Avoid including large-size files in the PR.
- [ ] Clean up long text outputs from code cells in the notebook.
- [ ] For security purposes, please check the contents and remove any sensitive info such as user names and private key.
- [ ] Ensure (1) hyperlinks and markdown anchors are working (2) use relative paths for tutorial repo files (3) put figure and graphs in the `./figure` folder
- [ ] Notebook runs automatically `./runner.sh -t <path to .ipynb file>`

### CODE REVIEW

File: deployment/ray/mednist_classifier_ray.ipynb [hunk 1]
```diff
@@ -41,7 +41,7 @@
    },
    "outputs": [],
    "source": [
-    "%pip install ray[serve]\n",
+    "!pip install \"ray[serve]\" \"protobuf==3.20.*\"\n",
```
Comment (mingxin-zheng) [inline deployment/ray/mednist_classifier_ray.ipynb:44]: Would it be better maintained in the future if we do "pip install ray --no-deps protobuf"? Thanks! @wyli

**Review by KumoLiu — APPROVED** [review]
Maybe we should restrict protobuf to version 3.20.1 seems there are other incompatibilities.
```
tensorboardx 2.5.1 requires protobuf<=3.20.1,>=3.8.0, but you have protobuf 3.20.3 which is incompatible.
onnx 1.12.0 requires protobuf<=3.20.1,>=3.12.2, but you have protobuf 3.20.3 which is incompatible.
```

File: deployment/ray/mednist_classifier_ray.ipynb (hunk 1 above)
Comment (mingxin-zheng) [reply deployment/ray/mednist_classifier_ray.ipynb:44]: You're right. I figure that `no-deps` is a flag that takes no arguments

**Comment by wyli** [comment]
this works fine with all the integration tests, I'm merging it for now

**Pull request merged**